starter-code/routes/signup.js

```diff
@@ -0,0 +1,35 @@
+const mongoose = require("mongoose");
+const bcrypt = require("bcrypt");
+const express = require("express");
+const User = require("../models/User");
+
+const authRoutes = express.Router();
+const Schema = mongoose.Schema;
+const bcryptSalt = 12;
+
+authRoutes.get("/signup", (req, res, next) => {
+  res.render("auth/signup");
+});
+
+authRoutes.post("/signup", (req, res, next) => {
+  console.log("DEBUG bcrypt", bcrypt);
+  let username = req.body.username;
+  let password = req.body.password;
+  var salt = bcrypt.genSaltSync(bcryptSalt);
+  var hashPass = bcrypt.hashSync(password, salt);
+
+  var newUser = User({
+    username,
+    password: hashPass
+  });
+
+  newUser.save(err => {
+    if (err) {
+      next(err);
+    } else {
+      res.redirect("/");
```

Comment:
This route seems to work well in saving a new user. The only thing is that you are redirecting the user to your root url `/`, which is currently not handled, so the user gets a 404 after signing up, even though he did sign up. Either redirect the user somewhere else (probably to a login view), or add a view to your root url.